libqtile/backend/wayland/core.py

```diff
@@ -406,6 +409,19 @@ def _on_new_pointer_constraint(self, _listener, wlr_constraint: PointerConstrain
     def _on_new_virtual_keyboard(self, _listener, virtual_keyboard: VirtualKeyboardV1):
         self._add_new_keyboard(virtual_keyboard.input_device)
 
+    def _on_new_inhibitor(self, _listener, idle_inhibitor: IdleInhibitorV1):
+        logger.debug("Signal: idle_inhibitor new_inhibitor")
+
+        for win in self.qtile.windows_map.values():
+            if isinstance(win, (window.Internal, window.Static)):
+                continue
+            assert isinstance(win, window.Window)
+
+            if win.surface.surface == idle_inhibitor.surface:
+                logger.error(str(win))
+                win.add_idle_inhibitor(idle_inhibitor)
+                break
```

Comment:
You can iterate through all of a windows surfaces with a callback like this:

```suggestion
        for win in self.qtile.windows_map.values():
            if isinstance(win, window.Window):
		win.surface.for_each_surface(win.add_idle_inhibitor, idle_inhibitor)
		if idle_inhibitor.data:
		    break
```
In this approach, you also need to adjust `Window.add_idle_inhibitor`:
```py
    def add_idle_inhibitor(self, surface: surface.Surface, _x, _y, inhibitor: IdleInhibitorV1) -> None:
        if surface == inhibitor.surface:
            inhibitor.data = self
            self._idle_inhibitors.append(inhibitor)
            self.add_listener(inhibitor.destroy_event, self._on_inhibitor_destroy)
            if len(self._idle_inhibitors) == 1:
                self.core.check_idle_inhibitor()
```
Then we are using `inhibitor.data` to know if the corresponding surface and window has been found.

Reply:
Nice, that works!